tiledb/sm/tile/generic_tile_io.h

```diff
@@ -122,14 +122,12 @@ class GenericTileIO {
    * information about the tile, and then reads the tile data. Note that it
    * creates a new Tile object with the header information.
    *
-   * @param tile The tile that will hold the read data.
    * @param file_offset The offset in the file to read from.
    * @param encryption_key The encryption key to use.
    * @param config The storage manager's config.
-   * @return Status
+   * @return Status, Tile
    */
-  Status read_generic(
-      Tile** tile,
+  tuple<Status, optional<tdb_unique_ptr<Tile>>> read_generic(
```

Comment:
Question: was there a reason to prefer `optional<unique_ptr<Tile>>` here over `optional<Tile>`? `load_data_from_generic_tile` returns `optional<Buffer>`, which seems like it should have the same cost as `optional<Tile>`. (IIUC Tile doesn't store anything inline, it either wraps a `unique_ptr<void*>`+deleter or a `std::vector<char>` via FilteredBuffer)

Reply:
Yes one caller has two branches calling this, and the resulting tile then gets used in common code. Having a unique pointer allows to not call the default constructor for Tile, which we'll want to remove at some point for C41 compliance.